resumablejs/resumable.d.ts

```diff
@@ -0,0 +1,374 @@
+     **/
+    chunkRetryInterval?: number;
+    /**
+     * Standard CORS requests do not send or set any cookies by default. In order to include cookies as part of the request, you need to set the withCredentials property to true. (Default: false)
+     **/
+    withCredentials?: boolean;
+  }
+  
+  class Resumable {
+    constructor(options:ConfigurationHash);
+    
+    /**
+     * A boolean value indicator whether or not Resumable.js is supported by the current browser.
+     **/
+    support: boolean;
+    /**
+     * A hash object of the configuration of the Resumable.js instance.
+     **/
+    opts: ConfigurationHash;
+    /**
+     * An array of ResumableFile file objects added by the user (see full docs for this object type below).
+     **/
+    files: ResumableFile[];
+	
+	defaults: ConfigurationHash;
+    
+    events: Event[];
+    version: number;
+    
+    /**
+     * Assign a browse action to one or more DOM nodes. Pass in true to allow directories to be selected (Chrome only).
+     **/
+    assignBrowse(domNode: Element, isDirectory: boolean): void;
+    assignBrowse(domNodes: Element[], isDirectory: boolean): void;
+    /**
+     * Assign one or more DOM nodes as a drop target.
+     **/
+    assignDrop(domNode: Element): void;
+    assignDrop(domNodes: Element[]): void;
+    unAssignDrop(domNode: Element): void;
+    unAssignDrop(domNodes: Element[]): void;
+    /**
+     * Start or resume uploading.
+     **/
+    upload(): void;
+    uploadNextChunk(): void;
+    /**
+     * Pause uploading.
+     **/
+    pause(): void;
+    /**
+     * Cancel upload of all ResumableFile objects and remove them from the list.
+     **/
+    cancel(): void;
+    fire(): void;
+    /**
+     * Returns a float between 0 and 1 indicating the current upload progress of all files.
+     **/
+    progress(): number;
+    /**
+     * Returns a boolean indicating whether or not the instance is currently uploading anything.
+     **/
+    isUploading(): boolean;
+    /**
+     * Add a HTML5 File object to the list of files.
+     **/
+    addFile(file: File, event: Event): void;
+    /**
+     * Cancel upload of a specific ResumableFile object on the list from the list.
+     **/
+    removeFile(file: ResumableFile): void;
+    /**
+     * Look up a ResumableFile object by its unique identifier.
+     **/
+    getFromUniqueIdentifier(uniqueIdentifier: string): void;
+    /**
+     * Returns the total size of the upload in bytes.
+     **/
+    getSize(): number;
+    getOpt(o: string): any;
+    
+    // Events
+	/**
+	 * Change event handler
+	 **/
+	handleChangeEvent(e: Event): void;
+	 
+	 /**
+	 * Drop event handler
+	 **/
+	handleDropEvent(e: Event): void;
+	
+    /**
+     * Listen for event from Resumable.js (see below)
+     **/
+    on(event: string, callback: Function): void;
```

Comment:
the most general overload should be last. Typescript resolves signatures in order. with this setup this is the overload that will always be picked.